packages/react-components/src/components/ConfigurationUserPreferencesStep.tsx

```diff
@@ -1,47 +1,191 @@
+    const [indexName, setIndexName] = useState('');
+    const [indexNames, setIndexNames] = useState<string[]>([]);
+    const [isIngestingMetadataOnly, setIsIngestingMetadataOnly] = useState(false);
+    const [isLoading, setIsLoading] = useState(false);
+
+    const handleTenantIdChange = (e): void => setTenantId(parseInt(e.target.value, 10));
+    const handleIndexNameChange = (e): void => setIndexName(e.target.value);
+    const handleIsIngestingMetadataChange = (e): void =>
+        setIsIngestingMetadataOnly(e.target.checked);
+
+    const handleSubmitUserPreferences = (): void => {
+        setIsLoading(true);
+        const severitiesFilter = selectedSeverities.map((severity) => severity.value).join(',');
+        const sourceTypesFilter = selectedSourceTypes
+            .map((sourceType) => sourceType.value)
+            .join(',');
+        saveConfiguration(
+            apiKey,
+            tenantId,
+            indexName,
+            isIngestingMetadataOnly,
+            severitiesFilter,
+            sourceTypesFilter
+        )
+            .then(() => {
+                setIsLoading(false);
+                toastManager.destroy(ToastKeys.ERROR);
+                toastManager.show({
+                    id: ToastKeys.SUCCESS,
+                    content: 'Configured Flare Account',
+                });
+                onUserPreferencesSaved();
+            })
+            .catch((e: any) => {
+                setIsLoading(false);
+                toastManager.show({
+                    id: ToastKeys.ERROR,
+                    isError: true,
+                    content: `Something went wrong. ${e.responseText}`,
+                });
+            });
+    };
+
+    useEffect(() => {
+        if (configurationStep === ConfigurationStep.UserPreferences) {
+            Promise.all([
+                fetchTenantId(),
+                fetchIngestMetadataOnly(),
+                fetchCurrentIndexName(),
+                fetchUserTenants(apiKey),
+                fetchFiltersSeverities(apiKey),
+                fetchFiltersSourceTypes(apiKey),
+                fetchAvailableIndexNames(),
+                fetchSeveritiesFilter(),
+                fetchSourceTypesFilter(),
+            ])
+                .then(
+                    ([
+                        id,
+                        ingestMetadataOnly,
+                        index,
+                        userTenants,
+                        availableSeverities,
+                        availableSourceTypeCategories,
+                        availableIndexNames,
+                        severitiesFilter,
+                        sourceTypesFilter,
+                    ]) => {
+                        setTenantId(id);
+                        setIsIngestingMetadataOnly(ingestMetadataOnly);
+                        setIndexName(index);
+                        if (id === -1 && userTenants.length > 0) {
```

Comment:
I was confused by the `-1`. I see that the `fetchTenantId()` function either returns parsed ID or `-1`. I guess my question is, why not simply `null`?  Null would make sense to me right off the hop, but I had to go and look and figure it out.

Reply:
Years of working with java primitives, the -1 was used by default since we weren't able to set it to null, I'll switch to undefined